simple-backend/stave_backend/lib/stave_config.py

```diff
@@ -0,0 +1,221 @@
+"""
+StaveConfig provides a basic interface to get/set configuration for Stave.
+"""
+
+import os
+import json
+import logging
+from enum import Enum
+from typing import Dict, Any, Iterator
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "StaveConfig"
+]
+
+
+class Fields(Enum):
+    """
+    Configurable fields of stave. Those do not start with '_' are treated
+    as public fields that can be showed and configured in CLI.
+    """
+    DB_FILE = "db_file"
+    LOG_FILE = "log_file"
+    DJANGO_SETTINGS_MODULE = "django_settings_module"
+    SECRET_KEY = "_secret_key"
+    DJANGO_SETTINGS = "_django_settings"
+
+    @classmethod
+    def itervalue(cls) -> Iterator[str]:
+        return (field.value for field in cls)
+
+
+class StaveConfig:
+    """
+    Provide basic getter/setter interfaces for all configurable paramaters
+    of Stave. Example usage:
+
+        sc = StaveConfig()
+        print(sc.db_file)
+        sc.log_file = "./log"
+
+    """
+
+    ADMIN_USERNAME = "admin"
+    ADMIN_PASSWORD = "admin"
+
+    CONFIG_PATH = os.path.join(os.path.expanduser('~'), ".stave")
+    CONFIG_FILE = os.path.join(CONFIG_PATH, "stave.conf")
+
+    DEFAULT_CONFIG_JSON = {
+        Fields.DJANGO_SETTINGS_MODULE.value: None,
+
+        # A basic django settings for Stave to run
+        Fields.DJANGO_SETTINGS.value: {
```

Comment:
We seem to store the whole Django config together with others with 'stave.conf'. Is this a bit confusing to users who are familiar with Django's "settings.py". How about we save the Django configs in its own file named "settings.py".